docker/docker.go

```diff
@@ -264,6 +265,30 @@ func WithSharedDirectory(hostPath, containerPath string) ConfigOption {
 }
 
 func withVolume(typ mount.Type, hostPath, containerPath string) ConfigOption {
+	c, err := client.NewEnvClient()
+	if err != nil {
+		logrus.Fatalf("could not create docker client")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	info, err := c.Info(ctx)
+	if err != nil {
+		logrus.Fatalf("could not get information")
+	}
+
+	isWinHost := info.OSType == "windows" || strings.Contains(
```

Comment:
I'd extract this functionality  in a separate function, like
```go
hostPath = mountSource(info, hostPath)
```
that would return the proper mount source for the passed OS